### PR TITLE
fix clang-tidy warnings

### PR DESCRIPTION
**Pull request by BenWibking**
### Description
This fixes many `clang-tidy` warnings thoughout the code. Most of them had to do with unnecessary header includes or const issues. No outputs should change at the bitwise level.

### Related issues
N/A

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [x] I have added a description (see above).
- [x] I have added a link to any related issues see (see above).
- [x] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [x] I have added tests for any new physics that this PR adds to the code.
- [x] I have tested this PR on my local computer and all tests pass.
- [x] I have manually triggered the GPU tests with the magic comment `/azp run`.
- [x] I have requested a reviewer for this PR.

### CODE REVIEW

**Comment by BenWibking** [comment]
The HIP failure looks like an AMD compiler bug.

**Review by markkrumholz — DISMISSED** [review]

**Comment by BenWibking** [comment]
There are lots of unexpected test failures. I'll dismiss the review and re-request a review when those are all fixed.

**Comment by BenWibking** [comment]
Ok, now the only remaining issue appears to be this ROCm compiler bug:
```console
In file included from /tmp/bwibking/_work/1/s/extern/amrex/Src/Base/AMReX_MultiFab.H:7:
In file included from /tmp/bwibking/_work/1/s/extern/amrex/Src/Base/AMReX_FArrayBox.H:6:
In file included from /tmp/bwibking/_work/1/s/extern/amrex/Src/Base/AMReX_BaseFab.H:19:
In file included from /tmp/bwibking/_work/1/s/extern/amrex/Src/Base/AMReX_Utility.H:15:
In file included from /tmp/bwibking/_work/1/s/extern/amrex/Src/Base/AMReX_Random.H:8:
In file included from /tmp/bwibking/_work/1/s/extern/amrex/Src/Base/AMReX_RandomEngine.H:11:
In file included from /opt/rocm-5.7.0/include/hiprand/hiprand.hpp:35:
In file included from /opt/rocm-5.7.0/include/hiprand/hiprand_kernel.h:64:
/opt/rocm-5.7.0/include/hiprand/hiprand_kernel_hcc.h:128:19: error: reference to 'detail' is ambiguous
    static_assert(detail::is_any_of<StateType,
                  ^
/opt/rocm-5.7.0/include/hiprand/hiprand_kernel_hcc.h:68:11: note: candidate found by name lookup is 'detail'
namespace detail {
          ^
/tmp/bwibking/_work/1/s/extern/amrex/Src/Base/AMReX_GpuComplex.H:425:11: note: candidate found by name lookup is 'amrex::detail'
namespace detail
          ^
```

**Comment by BenWibking** [comment]
Also a weird issue here: https://dev.azure.com/quokka-code/quokka-CI/_build/results?buildId=7522&view=logs&j=ab477ad6-9781-5b54-82ca-a0141b26726e&t=d1466d34-058c-541c-8cee-749bc5c3ebb3

**Comment by BenWibking** [comment]
Merging this is blocked on fixing a Microphysics issue (due to an underlying ROCm issue): https://github.com/AMReX-Astro/Microphysics/issues/1406

**Comment by psharda** [comment]
I am hoping this will also ensure the clang-tidy test passes for #368

**Comment by BenWibking** [comment]
> I am hoping this will also ensure the clang-tidy test passes for #368

We need Microphysics to avoid `using namespace amrex` before this PR will work on AMD.

Also, I think your clang-tidy issue is unrelated (unfortunately).

**Comment by psharda** [comment]
Hopefully, once #517 is merged, the AMD tests will also pass.

**Comment by BenWibking** [comment]
> Hopefully, once #517 is merged, the AMD tests will also pass.

No, we still need a fix in Microphysics for https://github.com/AMReX-Astro/Microphysics/issues/1406.

**Comment by BenWibking** [comment]
@psharda This PR will be fixed by this Microphysics PR: https://github.com/AMReX-Astro/Microphysics/pull/1465.

**Comment by BenWibking** [comment]
Depends on https://github.com/AMReX-Astro/Microphysics/pull/1474.

**Comment by BenWibking** [comment]
It's fixed now.

**Comment by BenWibking** [comment]
Fixed missing `std::` in planck_integral.hpp. Re-running tests...

File: src/EOS.hpp
Comment (psharda) [inline src/EOS.hpp:22]: Why remove these headers?

Comment (BenWibking) [reply src/EOS.hpp:22]: They're not used in this file.

**Review by psharda — APPROVED** [review]

**Pull request merged**
